### DISCUSSION TITLE
Phoenix desktop black screen on start up Arch linux

### DISCUSSION THREAD

**Discussion by kowalski713** [phcode-dev/phoenix#1506] [Q&A] [answered]
hey i used the installer to install the desktop version but it loads a black box with the header.  i have all the dependencies not sure what to do to fix it

**Comment by abose** [comment]
@charlypa arch linux issue.

**Comment by charlypa** [comment]
@kowalski713 
Is it a physical machine or virtual machine?

try updating the all dependencies by running `sudo pacman -Syu`

**Comment by kowalski713** [comment]
hey i tried updating but no luck i decided to try building from source and got this error
`** (phoenix-code-experimental-build:17151): WARNING **: 22:07:47.304: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
Appdata path is /home/kowalski/.local/share/io.phcode.dev
Directory created successfully /home/kowalski/.local/share/io.phcode.dev
Bundle ID is io.phcode.dev
Directory created successfully /home/kowalski/.local/share/io.phcode.dev
Bootconfig version is 1
AcceleratedSurfaceDMABuf was unable to construct a complete framebuffer`

im assuming some driver on the repository is out of date

**Reply by abose** [reply]
`AcceleratedSurfaceDMABuf was unable to construct a complete framebuffer`

This seems to be the issue. Can you help with the arch linux version, kernal version, and hardware details of graphics card, processor etc `hwinfo`.

**Comment by kowalski713** [comment]
```
cpu:
                       Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz, 4300 MHz
                       Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz, 4300 MHz
                       Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz, 4300 MHz
                       Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz, 4300 MHz
keyboard:
  /dev/input/event5    Microdia Backlit Gaming Keyboard
mouse:
  /dev/input/event7    Microdia Backlit Gaming Keyboard
                       GLORIOUS Model O V2 Wired Mouse
monitor:
                       ASUS VG328
graphics card:
                       nVidia GP104 [GeForce GTX 1060 6GB]
                       Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics Controller
sound:
                       Intel 9 Series Chipset Family HD Audio Controller
                       Intel Xeon E3-1200 v3/4th Gen Core Processor HD Audio Controller
                       nVidia GP104 High Definition Audio Controller
storage:
                       Intel 9 Series Chipset Family SATA Controller [AHCI Mode]
                       Sandisk Non-Volatile memory controller
network:
  eno1                 Intel Ethernet Connection I217-V
network interface:
  eno1                 Ethernet network interface
  lo                   Loopback network interface
disk:
  /dev/nvme0n1         Sandisk WD_BLACK SN770 2TB
  /dev/sdb             KINGSTON SV300S3
  /dev/sda             Samsung SSD 850
partition:
  /dev/nvme0n1p1       Partition
  /dev/nvme0n1p2       Partition
  /dev/sdb1            Partition
  /dev/sda1            Partition
usb controller:
                       Intel 9 Series Chipset Family USB EHCI Controller #2
                       Intel 9 Series Chipset Family USB EHCI Controller #1
                       Intel 9 Series Chipset Family USB xHCI Controller
bios:
                       BIOS
bridge:
                       Intel 82801 PCI Bridge
                       Intel 9 Series Chipset Family PCI Express Root Port 1
                       Intel Z97 Chipset LPC Controller
                       Intel Xeon E3-1200 v3/4th Gen Core Processor PCI Express x16 Controller
                       Intel 9 Series Chipset Family PCI Express Root Port 4
                       Intel 4th Gen Core Processor DRAM Controller
hub:
                       Intel Integrated Hub
                       Linux Foundation 2.0 root hub
                       Linux Foundation 3.0 root hub
                       Linux Foundation 2.0 root hub
                       Intel Hub
                       Linux Foundation 2.0 root hub
memory:
                       Main Memory
unknown:
                       FPU
                       DMA controller
                       PIC
                       Keyboard controller
  /dev/lp0             Parallel controller
                       Intel 9 Series Chipset Family ME Interface #1
                       Intel 9 Series Chipset Family SMBus Controller
                       Serial controller
```

Operating System: EndeavourOS 
KDE Plasma Version: 6.0.2
KDE Frameworks Version: 6.0.0
Qt Version: 6.6.2
Kernel Version: 6.8.2-arch1-1 (64-bit)
Graphics Platform: X11
Processors: 4 × Intel® Core™ i5-4690K CPU @ 3.50GHz
Memory: 23.3 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1060 6GB/PCIe/SSE2
Manufacturer: Gigabyte Technology Co., Ltd.
Product Name: Z97X-SLI

**Reply by abose** [reply]
Nice machine specs! Must be something with the specific arch distro. We have seen similar issues in VMs in VMware in sone distros, but not in any arch based distros till now.

**Comment by charlypa** [comment]
@kowalski713 
Thanks for sharing the system information. I installed the latest EndeavourOS and did an upgrade. However, looks like there is some issue with the driver. I too get a black screen.
To make Phoenix work we need only  webkit2gtk and gtk3 if these two packages are there in your system once the installer is run it should work in your distro as well.

**Comment by kowalski713** [comment]
i do have both those installed as well sorry and still black screen

**Reply by abose** [reply]
Looks like the specific distro is not supported due to some required graphics deps missing. @charlypa can fill you in on the exact details.

Our tests were mainly on the standard arch distribution at https://archlinux.org/download/ but typically we have seen that that should take care of most of the other flavors in debian/fedora.

**Comment by charlypa** [comment]
@kowalski713 
We use webGtk for rendering. Looks like due to some driver issues webgtk could not render Phoenix Code.  I am not able to figure out exactly what is missing from the driver. Thanks for bringing it up. we will take note of it am improve it further

**Reply by kowalski713** [reply]
Sounds good thanks for the response will check periodically for updates but no biggie the web version is great look forward to trying it again

**Comment by kowalski713** [comment, marked as answer]
hey so have found a work around from  [this issue](https://gitlab.archlinux.org/archlinux/packaging/packages/webkit2gtk/-/issues/1) for arch webkit2gtk package  

add WEBKIT_DISABLE_DMABUF_RENDERER=1 to /etc/environment

**Reply by abose** [reply]
Awesome!! How did you figure out the fix, just curious

**Reply by kowalski713** [reply]
i didnt figure it out i was just searching around the internet using the error (which pops up if building or running with terminal) `AcceleratedSurfaceDMABuf was unable to construct a complete framebuffer
` as my search keywords
 Then happened across the issue i linked in my other comment and the comments on there had this tip 
 original poster Downgraded webkit2gtk to 2.42.5-2  and seemed to resolve their issue as well
